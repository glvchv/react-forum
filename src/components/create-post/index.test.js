import React from 'react';
import CreatePost from '.';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
jest.mock('../error-message', () => 'ErrorMessage');

describe('Create Post component', () => {
    it('should render the component', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <CreatePost />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})