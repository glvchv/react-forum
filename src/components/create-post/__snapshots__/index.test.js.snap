// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Post component should render the component 1`] = `
<div
  className="wrapper"
>
  <div
    className="form"
  >
    <h1
      className="title"
    >
      What's on your mind?
    </h1>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <label
        className="label"
      >
        Post title:
      </label>
      <input
        onChange={[Function]}
        placeholder="e.g. What do you think about ...?"
        type="text"
      />
      <ErrorMessage
        error=""
      />
      <textarea
        className="text"
        onChange={[Function]}
      />
      <ErrorMessage
        error=""
      />
      <label
        className="label"
      >
        Category:
      </label>
      <select
        className="options"
        onChange={[Function]}
        value="entertainment"
      >
        <option
          value="sports"
        >
          Sports
        </option>
        <option
          value="entertainment"
        >
          Entertainment
        </option>
        <option
          value="education"
        >
          Education
        </option>
        <option
          value="technology"
        >
          Technology
        </option>
        <option
          value="other"
        >
          Other
        </option>
      </select>
      <button
        disabled={true}
        type="submit"
      >
        Post
      </button>
    </form>
  </div>
</div>
`;
